hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/S3Guard.java

```diff
@@ -242,7 +244,7 @@ public static S3AFileStatus putAndReturn(MetadataStore ms,
         if (status != null
             && s.getModificationTime() > status.getModificationTime()) {
           LOG.debug("Update ms with newer metadata of: {}", status);
-          ms.put(new PathMetadata(s));
+          S3Guard.putWithTtl(ms, new PathMetadata(s), timeProvider);
```

Comment:
nit: now this is being used, the javadoc of the timeProvider arg needs to be updated.

Reply:
what do you mean exactly?

Reply:
ah I see - you mean the javadoc above this method. got it!